hazelcast/src/main/java/com/hazelcast/map/impl/tx/TransactionalMapProxy.java

```diff
@@ -315,12 +319,9 @@ public void delete(Object key) {
     @SuppressWarnings("unchecked")
     public Set keySet(Predicate predicate) {
         checkTransactionState();
-        if (predicate == null) {
-            throw new NullPointerException("Predicate should not be null!");
-        }
-        if (predicate instanceof PagingPredicate) {
-            throw new NullPointerException("Paging is not supported for Transactional queries!");
-        }
+        checkNotNull(predicate, "Predicate should not be null!");
+        checkNotInstanceOf(PagingPredicate.class, predicate, "Paging is not supported for Transactional queries!");
```

Comment:
Yes.. I have not touched any of the behaviour. What is NPE --> NPE.. what is IllegalArgumentException --> remains IllegalArgumentException. 

That is why this is phase 1..